app/controllers/katello/api/v2/subscriptions_controller.rb

```diff
@@ -241,6 +254,14 @@ def index_organization
       return subscriptions
     end
 
+    def consumed_subscriptions(consumer, organization = current_organization)
+      entitlements = consumer.consumed_entitlements
+      entitlements.delete_if do |entitlement|
+        pool = Pool.find_pool(entitlement.poolId)
+        !pool.provider?(organization)
+      end
```

Comment:
Surprised we can't do something more like 'Pool.where(:provider => Provider.where(:organization => organization)).where(:id => entitlement.poolId)` ?